_plugins/sidebar.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: false
+
+require 'jekyll'
+require 'nokogiri'
+require 'json'
+
+module Jekyll
+  class Sidebar
+    attr_accessor :hash_pre_render
+    attr_accessor :filename_with_headers
+
+    def initialize
+      @hash_pre_render = {}
+      @filename_with_headers = {}
+    end
+
+    def pre_render(page)
+      menu_order = page['menu-order'].nil? ? 0 : page['menu-order']
+      url = page['url'].gsub('index.html', '').gsub('.html', '')
+      @hash_pre_render[url] = {
+        title: page['title'],
+        url: page['url'].gsub('.html', ''),
+        name: page['name'],
+        menu_order: menu_order
+      }
+    end
+
+    def post_write(site)
+      files = []
+      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+        doc = File.open(filename) { |f| Nokogiri::HTML(f) }
+        files.push(doc)
+        title = doc.title
+        url = ''
+        doc.xpath('/html/head/meta').each do |meta|
+          url = meta.attr('value') if meta.attr('property').eql? 'og:url'
+        end
+
+        headers = []
+        doc.xpath('//h2 ').each do |header|
+          next unless header['id']
+
+          child = header.last_element_child
+          header = {
+            id: header['id'],
+            title: header.content.strip,
+            url: "#{url.gsub('.html', '')}#{child['href']}"
+          }
+          headers.push(header)
+        end
+        sanitized_filename = sanitize_filename(filename)
+        @filename_with_headers[sanitized_filename] = { headers: headers }
+      end
+
+      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+        sanitized_filename = sanitize_filename(filename)
+        sidebar = render(sanitized_filename)
+        file = File.open(filename) { |f| Nokogiri::HTML(f) }
+        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
+          location.inner_html = sidebar
+        end
+        File.open(filename, 'w') { |f| f.write(file.to_xhtml(encoding: 'UTF-8')) }
+      end
+
+      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+    end
+
+    private
+
+    def sanitize_filename(filename)
+      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
+      sanitized_filename = sanitized_filename.gsub('index.html', '')
+      sanitized_filename.gsub('.html', '')
+    end
+
+    def generateSubgroup(filename, _key, value, has_subgroups)
```

Comment:
Codacy found an issue: [Perceived complexity for generateSubgroup is too high. [8/7]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=5659803)